impl/NotSoBasicLinearAlgebra.h

```diff
@@ -135,14 +143,19 @@ LUDecomposition<ParentType> LUDecompose(MatrixBase<ParentType, Dim, Dim, typenam
 
         if (j != argmax)
         {
-            for (int k = 0; k < Dim; ++k)
-            {
-                bla_swap(A(argmax, k), A(j, k));
-            }
+            auto row_argmax = A.Row(argmax);
+            auto row_j = A.Row(j);
+            Swap(row_argmax, row_j);
 
             decomp.parity = -decomp.parity;
 
-            bla_swap(idx[j], idx[argmax]);
+            // swap indicies
```

Comment:
```suggestion
            // swap indices
```
nit: funnily enough, I often make this exact same spelling error